File: CommunityCommutingApp-Backend/Trip_Management_Microservice/src/main/java/com/cognizant/cca/service/TripService.java
package com.cognizant.cca.service;

import java.util.List;
import com.cognizant.cca.model.Trip;
import com.cognizant.cca.response.ReturnResponse;

public interface TripService {
	public List<Trip> getAllRides(String rideStatus);
	public String addRide(Trip trip);
	public String updateRide(String tripId, Trip trip);
	public Trip getRide(String tripId);
//	public List<Trip> getFromLocAndToLocAndRideStatus(String fromLoc, String toLoc, String rideStatus);
	public List<Trip> getRideByRpId(String rpId);
	public ReturnResponse startRide(String tripId, Trip trip);
	public ReturnResponse endRide(String tripId, Trip trip);
	public Boolean deleteById(String tripId);
}

File: CommunityCommutingApp-Frontend/src/app/trip.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Trip } from './trip';
import { ReturnResponse } from './return-response';

@Injectable({
  providedIn: 'root'
})
export class TripService {

  constructor(private http : HttpClient) { }
  
  getAllTrip() : Observable<Trip[]> {
    return this.http.get<Trip[]>("http://localhost:8090/api/tripmanager/trips");
  }
  getAllTrips() : Observable<Trip[]> {
    return this.http.get<Trip[]>("http://localhost:8060/api/rideseeker/viewrides/list");
  }

  getTripByTripId(tripId: string) : Observable<Trip>{
      return this.http.get<Trip>("http://localhost:8090/api/tripmanager/" + tripId);
  }
  startTripById(tripId : string, trip  : Trip) :Observable<ReturnResponse>{
    return this.http.put<ReturnResponse>("http://localhost:8090/api/tripmanager/startride/"+tripId, trip);
  }
  endTripById(tripId : string, trip  : Trip) : Observable<ReturnResponse>{
      return this.http.put<ReturnResponse>("http://localhost:8090/api/tripmanager/endride/"+tripId, trip);
  }
}



// addRide(trip : Trip) : Observable<Trip> {
//   return this.http.post<Trip>("http://localhost:8080/api/tripmanager/new", trip);
// }
// 
// updateRide(tripId : string, trip : Trip) : Observable<Trip> {
//   return this.http.put<Trip>("http://localhost:8080/api/tripmanager/" + tripId + "/update", trip);
// }

// getRide(tripId : string) : Observable<Trip> {
//   return this.http.get<Trip>("http://localhost:8080/api/tripmanager/" + tripId);
// }